enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OVN Kubernetes network plugin uses OVN (https://www.ovn.org) to instantiate
+virtual networks for Kubernetes. These virtual networks use Geneve, a network
+virtualization overlay encapsulation protocol, to tunnel traffic across the
+underlay network between Kubernetes Nodes.
+
+IPsec is a protocol suite that enables secure network communications between
+IP endpoints by providing the following services:
+- Confidentiality
+- Authentication
+- Data Integrity
+
+OVN tunnel traffic is transported by physical routers and switches. These
+physical devices could be untrusted (devices in public network) or might be
+compromised. Enabling IPsec encryption for this tunnel traffic can prevent the
+traffic data from being monitored and manipulated.
+
+This enhancement allows cluster operators to enable traffic to be encapsulated
+by the IPsec protocol between worker nodes.
+
+## Motivation
+
+Encryption services are recommended when traffic is traversing an untrusted
+network. Encryption services may also be required for regulatory or compliance
+reasons.
+
+### Goals
+
+- Provide an option to enable IPsec encryption of all inter-worker node
+tunnel traffic across an entire Kubernetes cluster that has been configured with
+OVN Kubernetes.
+  - Assume that all nodes will be provisioned with certificates that have been
+  signed by the OpenShift Container Platform internal certificate authority
+  (CA).
+  - This configuration option will only be available on Geneve tunnels
+- All components used to implement this must be FIPs-compliant cryptographic
+components.
+- Provide an implementation that will be deployable on both Public and Private
+Cloud infrastructure. This will require that all inter-worker node traffic can
+be routed by the underlay network without reconfiguration of underlay firewall
+policies.
+- Reuse the same certificates, keys and CA used by the Cluster Management TLS
+implementation.
+
+### Non-Goals
+
+- Will not provide any Hardware Offload of the IPsec protocol. Although some
+checksums may be offloaded by standard NICs without any special enablement.
+- Will not provide any implementation for DPDK interfaces.
+- Will not provide an option to enable tunnels that have been configured with
+self-signed certificates or pre-shared keys.
+- Will not do any explicit performance optimization of the datapath. As the
+implementation of this enhancement will primarily require changes to the control
+and management path, it is expected that this will not alter the performance of
+the current OVS/Kernel IPsec implementation.
+- Will not provide any interface to allow a cluster operator to specify
+preferred encryption, integrity, authentication algorithms
+- Will not explicitly encrypt cluster management traffic. Cluster management
+traffic has already been encrypted using TLS.
```

Comment:
except the traffic that's on the Pod network, which we will be encrypting again (that's ok, just don't want to imply that we're doing something clever to not send some pod network traffic over the ipsec encrypted geneve tunnels)

Reply:
resolved elsewhere

Reply:
I am going to resolve this as it is the same theme as a comment above which I will use to track this.